src/Catalyst.Core.Modules.Kvm.Tests/IntegrationTests/DeltaExecutorTests.cs

```diff
@@ -43,344 +43,285 @@ namespace Catalyst.Core.Modules.Kvm.Tests.IntegrationTests
     public sealed class DeltaExecutorTests
     {
         private readonly ICryptoContext _cryptoContext = new FfiWrapper();
+        private CatalystSpecProvider _specProvider;
+        private StateProvider _stateProvider;
+        private StorageProvider _storageProvider;
+        private VirtualMachine _virtualMachine;
+        private IPublicKey _recipient;
+        private IPublicKey _sender;
+        private IPublicKey _poorSender;
+        private DeltaExecutor _executor;
+
+        public DeltaExecutorTests()
+        {
+            _specProvider = new CatalystSpecProvider();
```

Comment:
Yes I think it is a lot less repetitive like that, thanks for doing it. I would even have brought the `ITxTracer` up here too, but no big deal 👍

Reply:
makes more sense to refactor this later to be a test extending from the container provider and resolve these tested classes.